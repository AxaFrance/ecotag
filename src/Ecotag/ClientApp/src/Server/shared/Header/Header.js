import React from 'react';
import {Header, Name, User} from '@axa-fr/react-toolkit-all';
import logo from '@axa-fr/react-toolkit-core/dist/assets/logo-axa.svg';
import withAuthentication from '../../withAuthentication';
import i18next from "i18next";
import './Header.scss';

const formatRoles = (roles) => {
    return roles.reduce(
        (previousValue, currentValue) => {
            if (!previousValue) {
                return currentValue;
            }
            return `${previousValue}, ${currentValue}`;
        },
        ''
    );
}

export const AppHeader = ({user: {title = 'Ecotag', link = null, name = i18next.t('shared.header.errors.not_connected'), roles = ['NOT_FOUND']}}) => (
    <Header>
        <Name title={title} subtitle={i18next.t('shared.header.title')} img={logo} alt={title}/>
        <User name={name} href={link} profile={formatRoles(roles)}/>
    </Header>
);

export default withAuthentication()(AppHeader);
