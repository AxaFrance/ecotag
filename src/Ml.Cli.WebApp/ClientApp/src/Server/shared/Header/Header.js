import React from 'react';
import {Header, Name, User} from '@axa-fr/react-toolkit-all';
import logo from '@axa-fr/react-toolkit-core/dist/assets/logo-axa.svg';
import withAuthentication from '../../withAuthentication';
import './Header.scss';

const formatRoles = (roles) => {
    return roles.reduce(
        (previousValue, currentValue) => {
            if (!previousValue) {
                return currentValue;
            }
            return `${previousValue}, ${currentValue}`;
        },
        ''
    );
}

export const AppHeader = ({ user: { title = 'Ecotag', link = null, name = 'Non ConnectÃ©', roles = ['NOT_FOUND'] } }) => (
  <Header>
    <Name title={title} subtitle="An eco-system that aim to be eco for the planet" img={logo} alt={title} />
    <User name={name} href={link} profile={formatRoles(roles)} />
  </Header>
);

export default withAuthentication()(AppHeader);
