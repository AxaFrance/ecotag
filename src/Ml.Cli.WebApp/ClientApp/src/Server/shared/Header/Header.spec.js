import React from "react";
import '@testing-library/jest-dom';
import {render} from '@testing-library/react';
import {AppHeader} from "./Header";

describe('Header', () => {
    test('Render header page', async () => {
        
        const user = {
            roles:["youhou", "youhou"],
            name: 'Guillaume Chervet'
        }
        const { asFragment, container  } = render(<AppHeader user={user} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
