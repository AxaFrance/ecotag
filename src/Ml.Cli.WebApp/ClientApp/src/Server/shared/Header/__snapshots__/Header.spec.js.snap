// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Render header page 1`] = `
<DocumentFragment>
  <div
    class="af-header"
  >
    <div
      class="container af-header__wrapper"
    >
      <header
        class="af-header__content"
        role="banner"
      >
        <div
          class="af-header__name"
        >
          <a
            class="af-logo"
            href="/"
          >
            <img
              alt="Ecotag"
              class="af-logo__brand"
              src="logo-axa.svg"
            />
          </a>
          <h2
            class="af-header__title"
          >
            Ecotag
            <span
              class="af-header__subtitle"
            >
              An eco-system that aim to be eco for the planet
            </span>
          </h2>
        </div>
        <div
          class="af-info-user"
        >
          <span
            class="af-info-user__name"
          >
            Guillaume Chervet
          </span>
        </div>
      </header>
    </div>
  </div>
</DocumentFragment>
`;
